### PR TITLE
[Feat] layer duplication

### PR DESCRIPTION
**Pull request by b2kdaman**
Signed-off-by: Valentine <vpanchin@gmail.com>

### CODE REVIEW

File: src/reducers/vis-state-updaters.js [hunk 1]
```diff
@@ -933,6 +933,43 @@ export const removeLayerUpdater = (state, {idx}) => {
...
+/**
+ * duplicate layer
+ * @memberof visStateUpdaters
+ * @type {typeof import('./vis-state-updaters').duplicateLayerUpdater}
+ * @public
+ */
+export const duplicateLayerUpdater = (state, {idx}) => {
+  const {layers, layerData} = state;
+  const layerToDuplicate = cloneDeep(state.layers[idx]);
+  layerToDuplicate.id = `${layerToDuplicate.config.id}_copy`;
```
Comment (heshan0131) [inline src/reducers/vis-state-updaters.js:945]: this won't work if user copies the same layer twice. use `import {generateHashId} from '../utils/strings';` to get a new hash id for each duplicated layer

Comment (b2kdaman) [reply src/reducers/vis-state-updaters.js:945]: Got it

**Comment by b2kdaman** [comment]
@heshan0131 it's ready

**Comment by heshan0131** [comment]
rewrite in https://github.com/keplergl/kepler.gl/pull/1401

**Pull request closed without merging**